### ISSUE TITLE
Failing test - Integration tests for Agentd on Windows - Tier 0 and 1

### ISSUE THREAD

**Issue by sebasfalcone** [wazuh/wazuh#25603]
## Description
While performing the [post release tasks for 4.9.0](https://github.com/wazuh/wazuh/issues/25590), we found that the ["Integration tests for Agentd on Windows - Tier 0 and 1"](https://github.com/wazuh/wazuh/actions/runs/10725471692/job/29743644399?pr=25601) are failing. 
```
FAILED test_agentd/test_state/test_agentd_state.py::test_agentd_state[Successful_connection_with_remoted]
FAILED test_agentd/test_state/test_agentd_state.py::test_agentd_state[Only_remote_request_available]
=========== 2 failed, 35 passed, 26 warnings in 2540.24s (0:42:20) ============
```

> [!NOTE]
> It may be related to:
> - https://github.com/wazuh/wazuh/issues/23995

## DoD
- [ ] Root cause found
- [ ] Issue fixed

**Comment by LucioDonda** [comment]
This is a flaky test, [rerun](https://github.com/wazuh/wazuh/actions/runs/10725471692/job/29895090558) under the same conditions and results were OK.

### Successful_connection_with_remoted
`FAILED test_agentd/test_state/test_agentd_state.py::test_agentd_state[Successful_connection_with_remoted]`

When starting the log monitor called from _check_msg_count_ inside of _check_fields_ tries to convert current_value to int but the type doesn't match.
```
wazuh_log_monitor.start(only_new_events = True, callback=callbacks.generate_callback AGENTD_SENDING_AGENT_NOTIFICATION), accumulations = int(current_value))
E TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
test_agentd\utils.py:70: TypeError
```
Inside the test config for this case:

```
- name: "Successful_connection_with_remoted"
  description:
  configuration_parameters:
  metadata:
    remoted: true
    output:
    - type: "file"
      fields:
        status: "connected"
        last_keepalive: true
        last_ack: true
        msg_count: true
```


### Only_remote_request_available - WIP
`FAILED test_agentd/test_state/test_agentd_state.py::test_agentd_state[Only_remote_request_available]`

```
assert (wazuh_log_monitor.callback_result != None), f'Sending keep alive not found'
E AssertionError: Sending keep alive not found
test_agentd\utils.py:17: AssertionError
```

**Comment by LucioDonda** [comment]
Recheck with varying times with the internal configs:

[Result 1](https://github.com/wazuh/wazuh/actions/runs/10798470606/job/29952243339) 🟢 

* using timeout of 2 and 8 (above and below the default value)

[Result 2](https://github.com/wazuh/wazuh/actions/runs/10800200205/job/29957822435?pr=25676) 🔴 
and with string values and 100 (to force the error)

* checking type conversion (in yaml file it uses boolean but is obtained as string) :

```
    if expected_value == '':
        return expected_value == current_value
```

* forcing state_interval to 100

**Comment by LucioDonda** [comment]
# update 11/09

* rebase and recheck results with 4.9.1 [here](https://github.com/wazuh/wazuh/pull/25676/checks)
* local check with the same results.
   * analysis on the logs in progress.
* once a test fails all the subsequent cases fail too. This happens even when the logs, daemon, and anything that could be persisted between cases are cleaned or deleted.

**Comment by LucioDonda** [comment]
### update 12/09

* Still unable to replicate the failure under the same conditions.
  * tested on local environments or directly in [git hub actions](https://github.com/wazuh/wazuh/actions/runs/10725471692) with the same result.
  
![image](https://github.com/user-attachments/assets/b0b377a1-17a7-48e4-9435-f048325e8196)

* analyzing the failure cases in [GHA:](https://github.com/wazuh/wazuh/actions/workflows/integration-tests-agentd-tier-0-1-win.yml?query=is%3Afailure) 
  * In the last month there are 25 failure cases (from 112)
    * 10 are from an Install wazuh agent error (maybe caused by a feature in progress).
    * 5 where skipped
    * 3 were failures because of the code that was being modified.
    * 2 forced by me
    * In 2 the error is caused by not being able to check the existence of the _last_keepalive_ key in the state file
    * the rest of them only 2 repeat the accumulate error:
```
    >       wazuh_log_monitor.start(only_new_events = True, callback=callbacks.generate_callback(AGENTD_SENDING_AGENT_NOTIFICATION), accumulations = int(current_value))
E       TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
```

* Comparing GHA raw logs on failure cases and success ones there are no noticeable differences besides the errors at the end.

## TL;DR:
* IMO It won't be easy to replicate the error as it was first showed on the issue.
* We can work on adding some messages regarding file cleanup and reload of services, this can help in future cases the test fails.
* Right now we can't see any issue or problem related to agents that could be linked to this error.

**Comment by vikman90** [comment]
@LucioDonda Thanks for your effort with this issue.


Here’s the response in English, adapted for a professional tone suitable for a GitHub comment:

I assume the analysis was done by reading the agent's log. Do we have it available? It would be useful to have the _ossec.log_ exported as an artifact.

Do you recall Gabi's comment the other day? We want to avoid needing to manually reproduce issues by interacting with the test when it fails. In this case, it's clear that we don't have the exact conditions to trigger the failure, which is leading to excessive time being spent on this.

One possible suspicion is that we might be dealing with a race condition in the _ossec.log_ file writing. However, I don’t believe relying solely on the log to infer behavior will yield reliable results.

Cheers!

**Comment by vikman90** [comment]
@LucioDonda Thanks for your effort with this issue.

I assume the analysis was done by reading the agent's log. Do we have it available? It would be useful to have the ossec.log exported as an artifact.

One possible suspicion is that we might be dealing with a race condition in the ossec.log file writing. However, I don’t believe relying solely on the log to infer behavior will yield reliable results.

Cheers!

**Comment by vikman90** [comment]
We have decided to close this issue. The method the test uses to detect keep-alives is invalid.

If it is detected again, we will open an issue to abort the test. When we do new Agent tests, we will use mocks (something like Remoted-Simulator) for similar cases.
